coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala

```diff
@@ -34,6 +34,7 @@ object LogicalPlanUtils extends StrictLogging {
    */
   def hasDescendantAggregate(lp: LogicalPlan): Boolean = lp match {
     case _: Aggregate                 => true
+    case bj: BinaryJoin                => bj.operator == BinaryOperator.LUnless
```

Comment:
What about e.g. `foo{} and bar{}`? If that is pushed down and `bar` exists on only one partition, then it's possible there will be rows missing from the final result.

I think the general case should be okay:
```
case bj: BinaryJoin => true
```
(as long as there's some performance testing involved).